lib/data-types.js

```diff
@@ -460,6 +460,42 @@ class TIME extends ABSTRACT {
   toSql() {
     return 'TIME';
   }
+  validate(value) {
+    if (!Validator.isDate(String(value)) && !Validator.isDate(`01-01-1970 ${String(value)}`)) {
+      throw new sequelizeErrors.ValidationError(util.format('%j is not a valid date', value));
+    }
+    return true;
+  }
+  _stringify(date) {
+    if (date instanceof Date) {
+      return moment(date).format('HH:mm:ss');
+    } 
+    return date;
+  }
+  _sanitize(value, options) {
+    if ((!options || !options.raw) && value instanceof Date ) {
+      return moment(value).format('HH:mm:ss');
+    }
+    return value;
+  }
+  _isChanged(value, originalValue) {
+    if (
+      originalValue && 
+      !!value &&
+        (
+          value === originalValue ||
+          value instanceof Date && originalValue instanceof Date && 
+          Math.floor(value.getTime() / 1000) === Math.floor(originalValue.getTime() / 1000)
```

Comment:
This should not exclude milliseconds, TIME supports milliseconds

Reply:
I think milliseconds support is dependent on the dialect, but I will take a look into that

Reply:
That's a good point, thank you for double-checking :)

Reply:
I researched this issue and it seems like Mysql, Mariadb, SQLite, Postgres and MSSQL support HH:MM:SS.sss formatted times. So I have changed the code to save milliseconds also in case a date object is passed as the argument.